architecture.adoc

```diff
@@ -0,0 +1,134 @@
+= Architecture
+
+= Introduction
+
+Mokey is an application that collects data about running applications on an Aurora OpenShift cluster and exposes that
+information via a ReST API in a cached and fast-to-access manner.
+
+The collection process is scheduled and parallelized, and information about the applications is updated in real time as
+the process progresses.
+
+This document is a description of the overall architecture of the module, divided into two main sections;
+<<core-concepts, the Core Concepts>> and <<api-resources, the API resources>>, describing the internal aspects of the
+module, and the externally exposed resources, respectively.
+
+
+[[core-concepts]]
+= Core Concepts (Internal)
+
+This section contains a description of the core concepts of the module, with implementation and domain model details.
+It does not contain any details of the resources that are exposed from the API, only the classes and objects that are
+used to support the internal data collection workflow; crawling.
+
+[[reference-data]]
+== Reference Data
+
+Classes and objects that represent static data and rules.
+
+Package: no.skatteetaten.aurora.mokey.model
+
```

Comment:
I feel that NFR should be handled outside of Mokey. Mokey should provide raw data to enable this but not the rules themselves.

We do not have to check for NFRViolations every time we run the crawl. 

There are lots of external services that must be contacted to do NFR check. Like splunk/prometheus aso.

Reply:
Where should the rules be placed? should this be a separate service?

Reply:
A seperate service would be my ideal choice. Then it can run in an individual schedule and fetch information from both mokey and other sources like splunk/prometheus/tracing.